src/commands/flinkStatements.test.ts

```diff
@@ -133,58 +131,64 @@ describe("commands/flinkStatements.ts", () => {
     });
   });
 
-  describe("fireEmitterWhenFlinkStatementIsCreatingFunction", () => {
+  describe("fire emitter when the Flink Statement creates a function", () => {
+    const database = TEST_CCLOUD_KAFKA_CLUSTER as CCloudFlinkDbKafkaCluster;
+    let waitForStatementCompletionStub: sinon.SinonStub;
+    let waitForResultsFetchableStub: sinon.SinonStub;
+    let openFlinkStatementResultsViewStub: sinon.SinonStub;
+    const createFuncStatement = createFlinkStatement({
+      sqlStatement:
+        "CREATE FUNCTION `testFunction` AS 'com.test.TestClass' USING JAR 'confluent-artifact://artifact-id';",
+    });
+    let stubbedUDFsChangedEmitter: sinon.SinonStubbedInstance<
+      vscode.EventEmitter<CCloudFlinkDbKafkaCluster>
+    >;
+
+    beforeEach(() => {
+      waitForStatementCompletionStub = sandbox.stub(statementUtils, "waitForStatementCompletion");
+      waitForResultsFetchableStub = sandbox.stub(statementUtils, "waitForResultsFetchable");
+      openFlinkStatementResultsViewStub = sandbox.stub(statements, "openFlinkStatementResultsView");
+      stubbedUDFsChangedEmitter = eventEmitterStubs(sandbox).udfsChanged!;
+    });
+
     it("should fire the emitter", async () => {
-      const statement = createFlinkStatement({
-        sqlStatement:
-          "CREATE FUNCTION `testFunction` AS 'com.test.TestClass' USING JAR 'confluent-artifact://artifact-id';",
-      });
-      statement.status = {
-        ...statement.status,
+      createFuncStatement.status = {
+        ...createFuncStatement.status,
         phase: Phase.COMPLETED,
         traits: {
           sql_kind: "CREATE_FUNCTION",
         },
       };
-      const database = TEST_CCLOUD_KAFKA_CLUSTER as CCloudFlinkDbKafkaCluster;
-      const stubbedEventEmitters = eventEmitterStubs(sandbox);
-      const waitForStatementCompletionStub = sandbox.stub(
-        statementUtils,
-        "waitForStatementCompletion",
-      );
-      waitForStatementCompletionStub.resolves(statement);
-      const stubbedUDFsChangedEmitter = stubbedEventEmitters.udfsChanged!;
-      await fireEmitterWhenFlinkStatementIsCreatingFunction(statement, database);
+      waitForResultsFetchableStub.resolves();
+      openFlinkStatementResultsViewStub.resolves();
```

Comment:
Minor: we don't need these explicit `.resolves()` unless we want to change some default behavior or explicitly control its return value (like the `.resolves(createFuncStatement)` below)